src/lib/transaction_logic/parties_logic.ml

```diff
@@ -1510,9 +1511,8 @@ module Make (Inputs : Inputs_intf) = struct
     in
     Bool.(assert_ (not (is_start' &&& global_excess_update_failed))) ;
     let local_state =
-      { local_state with
-        success = Bool.(local_state.success &&& not global_excess_update_failed)
-      }
+      Local_state.add_check local_state Global_excess_overflow
+        Bool.(not overflow)
```

Comment:
This should be `Bool.(not global_excess_update_failed)`, we should only check the excess' overflow when we're using the result.